common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -270,6 +279,123 @@ public byte[] getBytes() {
     }
   }
 
+  /**
+   * Utility methods and constants for UTF-8 string validation.
+   */
+
+  private static boolean isValidContinuationByte(byte b) {
+     return (byte) 0x80 <= b && b <= (byte) 0xBF;
+  }
+
+  private static boolean isValidSecondByte(byte b, byte firstByte) {
+    return switch (firstByte) {
+      case (byte) 0xE0 -> (byte) 0xA0 <= b && b <= (byte) 0xBF;
+      case (byte) 0xED -> (byte) 0x80 <= b && b <= (byte) 0x9F;
+      case (byte) 0xF0 -> (byte) 0x90 <= b && b <= (byte) 0xBF;
+      case (byte) 0xF4 -> (byte) 0x80 <= b && b <= (byte) 0x8F;
```

Comment:
```suggestion
      case (byte) 0xE0 -> b >= (byte) 0xA0 && b <= (byte) 0xBF;
      case (byte) 0xED -> b >= (byte) 0x80 && b <= (byte) 0x9F;
      case (byte) 0xF0 -> b >= (byte) 0x90 && b <= (byte) 0xBF;
      case (byte) 0xF4 -> b >= (byte) 0x80 && b <= (byte) 0x8F;
```
uber nit: same comment here.